In [2]:
import numpy as np
import pandas as pd
names = ['year','','','','','gender','','','headcount','','','','dept','major']
df = pd.read_csv('degrees.csv',names=names,header=0,usecols=[0,5,8,12,13])

In [2]:
df.head()

year            gender  headcount                      dept  \
0  1983-84  Decline to State          1  African American Studies   
1  1983-84  Decline to State          1  African American Studies   
2  1983-84  Decline to State          9  Ag & Resource Econ & Pol   
3  1983-84              Male          1  Ag & Resource Econ & Pol   
4  1983-84  Decline to State          8              Anthropology   

                       major  
0   Afr Amer Stds-Humanities  
1   Afr Amer Stds-Social Sci  
2  Pol Econ of Nat Resources  
3  Pol Econ of Nat Resources  
4               Anthropology

In [21]:
# Filter to most recent year
df2 = df[df['year'] == '2015-16']
# Calculate department aggregates
agg = df2.groupby(['dept', 'gender']).sum().div(df2.groupby(['dept']).sum())
# Save to json
agg.to_csv('graphic/csv/2015_16_by_gender.csv')
agg.head()

headcount
dept                     gender           
African American Studies Female   0.500000
                         Male     0.500000
Ag & Resource Econ & Pol Female   0.554745
                         Male     0.445255
Anthropology             Female   0.732143

In [7]:
agg.sort_values('headcount', ascending=False)

headcount
dept                           gender           
Other Env Design Programs      Female   1.000000
Scandinavian                   Female   1.000000
Near Eastern Studies           Female   0.928571
Gender & Womens Studies        Female   0.900000
Electrical Eng & Computer Sci  Male     0.887671
Nuclear Engineering            Male     0.875000
L&S Envir Econ & Policy        Male     0.857143
History of Art                 Female   0.842105
L&S Social Welfare             Female   0.839695
Mechanical Engineering         Male     0.837838
Physics                        Male     0.800000
Spanish & Portuguese           Female   0.794872
Chem & Biomolecular Eng        Male     0.786885
Comparative Literature         Female   0.777778
L&S Public Health              Female   0.770751
Psychology                     Female   0.770588
Art Practice                   Female   0.761905
Italian Studies                Female   0.750000
Slavic Languages & Literatures Female   0.750000
Astronomy                      Male     0.735294
Anthropology                   Female   0.732143
Env Sci, Policy, & Mgmt        Female   0.728302
L&S Computer Science           Male     0.715640
Nutritional Sciences & Tox     Female   0.714286
Sociology                      Female   0.714286
Integrative Biology            Female   0.701754
City & Regional Planning       Female   0.697674
Mathematics                    Male     0.683019
French                         Female   0.666667
Theater Dance & Perf Stds      Female   0.666667
...                                          ...
UG Interdisciplinary Studies   Male     0.339207
English                        Male     0.338182
Theater Dance & Perf Stds      Male     0.333333
French                         Male     0.333333
Mathematics                    Female   0.316981
City & Regional Planning       Male     0.302326
Integrative Biology            Male     0.298246
Nutritional Sciences & Tox     Male     0.285714
Sociology                      Male     0.285714
L&S Computer Science           Female   0.284360
Env Sci, Policy, & Mgmt        Male     0.271698
Anthropology                   Male     0.267857
Astronomy                      Female   0.264706
Italian Studies                Male     0.250000
Slavic Languages & Literatures Male     0.250000
Art Practice                   Male     0.238095
Psychology                     Male     0.229412
L&S Public Health              Male     0.229249
Comparative Literature         Male     0.222222
Chem & Biomolecular Eng        Female   0.213115
Spanish & Portuguese           Male     0.205128
Physics                        Female   0.200000
Mechanical Engineering         Female   0.162162
L&S Social Welfare             Male     0.160305
History of Art                 Male     0.157895
L&S Envir Econ & Policy        Female   0.142857
Nuclear Engineering            Female   0.125000
Electrical Eng & Computer Sci  Female   0.112329
Gender & Womens Studies        Male     0.100000
Near Eastern Studies           Male     0.071429

[130 rows x 1 columns]

In [132]:
agg.loc['Electrical Eng & Computer Sci']

headcount
gender           
Female   0.112329
Male     0.887671

In the 2015-16 academic year, 89% of degrees awarded by the EECS department went to male students. That's the highest proportion of male graduates of any department.

In [133]:
agg.loc['L&S Computer Science']

headcount
gender           
Female    0.28436
Male      0.71564

In the 2015-16 academic year, 72% of degrees awarded by the L&S CS department went to male students.

In [155]:
df3 = df2[df2['dept'].str.contains('Computer')]
agg3 = df3.groupby(['gender']).sum()
agg3

headcount
gender           
Female        161
Male          626

In [158]:
df3.sum(0).headcount

787

In [159]:
agg3.loc['Male'].headcount/df3.sum(0).headcount

0.795425667090216

In total, 80% of the 787 degrees awarded in computer science (EECS and L&S CS) in the 2015-16 academic year went to males.

In [19]:
# Back to considering all years
# Filter to EECS & CS
df4 = df[df['dept'].str.contains('Computer')]
# Get rid of 1983-84, since it contains an unsually high number of students declining to state their gender
df4 = df4[df4['year'] != '1983-84']
#agg4 = df4[df4['gender'] == 'Male'].groupby(['year']).sum().div(df4.groupby(['year']).sum())
agg4 = df4.groupby(['year','gender']).sum().div(df4.groupby(['year']).sum())
# Save to json
agg4.to_csv('graphic/csv/cs_by_year.csv')
agg4.to_json('graphic/json/cs_by_year.json')
agg4

headcount
year    gender                     
1984-85 Decline to State   0.031546
        Female             0.252366
        Male               0.716088
1985-86 Decline to State   0.011364
        Female             0.201705
        Male               0.786932
1986-87 Decline to State   0.008451
        Female             0.225352
        Male               0.766197
1987-88 Female             0.248447
        Male               0.751553
1988-89 Female             0.188679
        Male               0.811321
1989-90 Female             0.186916
        Male               0.813084
1990-91 Decline to State   0.007117
        Female             0.174377
        Male               0.818505
1991-92 Decline to State   0.006289
        Female             0.169811
        Male               0.823899
1992-93 Female             0.184874
        Male               0.815126
1993-94 Female             0.171171
        Male               0.828829
1994-95 Decline to State   0.003021
        Female             0.190332
        Male               0.806647
1995-96 Female             0.184573
        Male               0.815427
...                             ...
2001-02 Female             0.197826
        Male               0.802174
2002-03 Female             0.184685
        Male               0.815315
2003-04 Female             0.191489
        Male               0.808511
2004-05 Female             0.172222
        Male               0.827778
2005-06 Female             0.161290
        Male               0.838710
2006-07 Female             0.139130
        Male               0.860870
2007-08 Female             0.105769
        Male               0.894231
2008-09 Female             0.101156
        Male               0.898844
2009-10 Female             0.102273
        Male               0.897727
2010-11 Female             0.100796
        Male               0.899204
2011-12 Female             0.138498
        Male               0.861502
2012-13 Female             0.159596
        Male               0.840404
2013-14 Female             0.136000
        Male               0.864000
2014-15 Female             0.182209
        Male               0.817791
2015-16 Female             0.204574
        Male               0.795426

[72 rows x 1 columns]

The proportion of male students receiving degrees in EECS and L&S CS rose throughout the 2000s, peaked in 2010-11 at 90%, and has declined to 80% since then.

In [186]:
df4 = df[df['dept'].str.contains('Electrical Eng')]
agg4 = df4[df4['gender'] == 'Male'].groupby(['year']).sum().div(df4.groupby(['year']).sum())
agg4

headcount
year              
1983-84   0.642857
1984-85   0.777778
1985-86   0.836910
1986-87   0.784810
1987-88   0.796296
1988-89   0.842424
1989-90   0.852814
1990-91   0.827586
1991-92   0.861244
1992-93   0.856000
1993-94   0.892704
1994-95   0.839216
1995-96   0.859504
1996-97   0.857143
1997-98   0.853556
1998-99   0.851986
1999-00   0.816993
2000-01   0.818750
2001-02   0.838710
2002-03   0.855769
2003-04   0.796353
2004-05   0.804598
2005-06   0.842424
2006-07   0.853846
2007-08   0.906250
2008-09   0.906367
2009-10   0.906716
2010-11   0.919231
2011-12   0.887460
2012-13   0.881818
2013-14   0.896806
2014-15   0.861035
2015-16   0.887671

In [4]:
# Get rid of 1983-84, since it contains an unsually high number of students declining to state their gender
df5 = df[df['year'] != '1983-84']
agg5 = df5[df5['gender'] == 'Female'].groupby(['year','dept']).sum()
agg5.columns = ['female_count']
agg5

agg6 = df5.groupby(['year','dept']).sum()
agg6.columns = ['total_count']
agg6

agg7 = pd.concat([agg5, agg6], axis=1)
agg7['female_pct'] = agg7['female_count'].div(agg7['total_count'])
agg7
agg7.to_csv('site/csv/data.csv')